keysign/bluetoothoffer.py

```diff
@@ -0,0 +1,143 @@
+    logging.getLogger().error("You seem to be trying to execute " +
+                              "this script directly which is discouraged. " +
+                              "Try python -m instead.")
+    import os
+    parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    os.sys.path.insert(0, parent_dir)
+    os.sys.path.insert(0, os.path.join(parent_dir, 'monkeysign'))
+    import keysign
+    __package__ = str('keysign')
+
+from .gpgmh import get_public_key_data, get_usable_keys
+from .util import get_local_bt_address, mac_generate
+
+log = logging.getLogger(__name__)
+
+
+class BluetoothOffer:
+    def __init__(self, key, port=3, size=1024):
+        self.key = key
+        self.port = port
+        self.size = size
+        self.server_socket = None
+        self.message_def = None
+        self.stopped = False
+        self.code = None
+
+    @inlineCallbacks
+    def start(self):
+        self.stopped = False
+        message = "Back"
+        success = False
+        try:
+            while not self.stopped and not success:
+                # server_socket.accept() is not stoppable. So with select we can call accept()
+                # only when we are sure that there is already a waiting connection
+                ready_to_read, ready_to_write, in_error = yield threads.deferToThread(
+                    select.select, [self.server_socket], [], [], 0.5)
+                if ready_to_read:
+                    # We are sure that a connection is available, so we can call
+                    # accept() without deferring it to a thread
+                    client_socket, address = self.server_socket.accept()
+                    key_data = get_public_key_data(self.key.fingerprint)
+                    kd_decoded = key_data.decode('utf-8')
+                    yield threads.deferToThread(client_socket.sendall, kd_decoded)
+                    log.info("Key has been sent")
+                    success = True
+                    message = None
+        except Exception as e:
+            log.error("An error occurred: %s" % e)
+            success = False
+            message = e
+
+        returnValue((success, message))
+
+    def allocate_code(self):
+        try:
+            code = get_local_bt_address().upper()
+        except dbus.exceptions.DBusException as e:
+            if e.get_dbus_name() == "org.freedesktop.systemd1.NoSuchUnit":
+                log.info("No Bluetooth devices found, probably the bluetooth service is not running")
+            elif e.get_dbus_name() == "org.freedesktop.DBus.Error.UnknownObject":
+                log.info("No Bluetooth devices available")
+            else:
+                log.error("An unexpected error occurred %s", e.get_dbus_name())
+            self.code = None
+            return None
+        if self.server_socket is None:
+            self.server_socket = BluetoothSocket(RFCOMM)
+            self.server_socket.bind(("", 0))
```

Comment:
Fixed with 34ffa3a

Right now I leaved the BDADDR_ANY and added a comment about the address binding.